### PR TITLE
[api-minor] Let `Catalog.getAllPageDicts` return an *empty*  dictionary when loading the first /Page fails (issue 15590)

### LINKED ISSUE #15590: Broken pdf with a script
Attach (recommended) or Link to PDF file here:
[contains-script.pdf](https://github.com/mozilla/pdf.js/files/9814103/contains-script.pdf)

Configuration:
- Web browser and its version: Firefox nightly
- Operating system and its version: Windows 11

I found this pdf here:
https://twitter.com/ericlaw/status/1582411655874347019

it's likely a hand-written one, it opens correctly in Acrobat and Chrome.

### PR DESCRIPTION
**Pull request by Snuffleupagus**
*Please refer to the individual commit messages for additional details.*

Note that while the *first* patch feels fairly uncontroversial, the *second* one I'm a little less "happy" about myself since it'll lead to us displaying nothing but an empty page in some corrupt documents (that we previously just failed to open).

### CODE REVIEW

File: src/core/catalog.js [hunk 1]
```diff
@@ -1190,7 +1191,9 @@ class Catalog {
    * Eagerly fetches the entire /Pages-tree; should ONLY be used as a fallback.
    * @returns {Promise<Map>}
    */
-  async getAllPageDicts(recoveryMode = false) {
+  async getAllPageDicts(recoveryMode = false, evaluatorOptions = null) {
```
[the remark this answers is not in the document]
Comment (Snuffleupagus) [reply src/core/catalog.js]: Really good point; this prompted me look through the patch again, only to realize that we already had access this data here and could remove all of this parameter passing :-)

**Review by calixteman — APPROVED** [review]
LGTM.

File: src/core/xref.js [hunk 2]
```diff
@@ -545,7 +545,29 @@ class XRef {
...
+        // Attempt to handle (some) corrupt documents, where no 'startxref'
+        // operators are present (fixes issue15590.pdf).
+        if (position + contentLength >= length) {
+          const endPos = position + skipUntil(buffer, position, objBytes) + 4;
+
+          const checkPos = Math.max(endPos - CHECK_CONTENT_LENGTH, position);
+          const tokenStr = bytesToString(buffer.subarray(checkPos, endPos));
+
+          // Find the first "obj" occurrence after the 'trailer' operator.
+          const objToken = nestedObjRegExp.exec(tokenStr);
```
[the remark this answers is not in the document]
Comment (Snuffleupagus) [reply src/core/xref.js:559]: Perhaps not, but given that we're parsing *arbitrary* stream data it just seemed a little bit safer to include that part to reduce the risk of any false positives.

**Pull request merged**